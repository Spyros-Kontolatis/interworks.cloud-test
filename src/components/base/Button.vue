<template>
  <button class="border border-primary relative" :class="[...buttonClasses]">
    <Loader v-if="loading" class="w-8 h-8 top-3 left-2" />

    <slot></slot>
  </button>
</template>

<script setup lang="ts">
import { defineProps, computed } from "vue";
import {
  sizes,
  textColors,
  bgColors,
  rounded,
  outline,
} from "@/config/components/base/button";
import type { ButtonProps } from "@/types/components/base/button";
import Loader from "./Loader.vue";

const props = defineProps<ButtonProps>();

const buttonClasses = computed(() => {
  return [
    bgColors[props.color],
    textColors[props.color],
    sizes[props.size],
    props.rounded && rounded[props.size],
    props.outline && outline[props.color],
    (props.disabled || props.loading) &&
      "opacity-50 cursor-not-allowed pointer-events-none",
  ];
});
</script>
