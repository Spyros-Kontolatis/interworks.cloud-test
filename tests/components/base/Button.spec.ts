import { shallowMount } from "@vue/test-utils";
import Button from "../../../src/components/base/Button.vue";

describe("Button.vue", () => {
  it("renders the component with required props", () => {
    const wrapper = shallowMount(Button, {
      props: {
        size: "sm",
        color: "primary",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("renders the different sizes correctly ", () => {
    const wrapper = shallowMount(Button, {
      props: {
        size: "xs",
        color: "primary",
      },
    });
    expect(wrapper.find(".p-2.text-xs.rounded-md")).toBeDefined();
    wrapper.setProps({
      color: "primary",
      size: "sm",
    });
    expect(wrapper.find(".p-2.text-sm.rounded-md")).toBeDefined();
    wrapper.setProps({
      color: "primary",
      size: "md",
    });
    expect(wrapper.find(".p-4.text-md.rounded-xl")).toBeDefined();
    wrapper.setProps({
      color: "primary",
      size: "lg",
    });
    expect(wrapper.find(".p-2.text-lg.rounded-xl")).toBeDefined();
    wrapper.setProps({
      color: "primary",
      size: "xl",
    });
    expect(wrapper.find(".p-4.text-lg.rounded-2xl")).toBeDefined();
  });
  it("renders the different color schemes correctly ", () => {
    const wrapper = shallowMount(Button, {
      props: {
        size: "xs",
        color: "primary",
      },
    });
    expect(wrapper.find(".text-white.bg-primary")).toBeDefined();
    wrapper.setProps({
      color: "secondary",
      size: "sm",
    });
    expect(wrapper.find(".text-black.bg-secondary")).toBeDefined();
  });
});
