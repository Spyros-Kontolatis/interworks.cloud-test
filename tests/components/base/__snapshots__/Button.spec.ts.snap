// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders the component with required props 1`] = `
<button
  class="border border-primary relative bg-primary text-white p-2 text-sm"
>
  <!--v-if-->
  
  
</button>
`;
